### PR TITLE
Feat/enable disable poi

### PR DESCRIPTION
**Pull request by Scttpr**

### CODE REVIEW

**Comment by Scttpr** [comment]
[Activer / Désactiver le POI](https://trello.com/c/GzerBjy5/257-activer-d%C3%A9sactiver-le-poi)

File: routes/pois.mjs [hunk 1]
```diff
@@ -50,6 +75,9 @@ const router = generateCRUD(Poi, {
     ],
     async main(ctx) {
       const { offset, limit } = ctx.parseRangePagination(Poi, { max: 1000 });
+
+      await Poi.processDocumentsToAddEnable();
```
Comment (Nainterceptor) [inline routes/pois.mjs]: NOPE

We don't need to change all documents everytime !

File: routes/pois.mjs [hunk 2]
```diff
@@ -30,8 +30,33 @@ const router = generateCRUD(Poi, {
...
+        await next();
+      },
+      async (ctx, next) => {
+        const filters = Object.keys(ctx.filters).map((key) => ({
+          [key]: ctx.filters[key],
+        }));
+
+        if (filters.length > 0) {
+          ctx.filters = {
+            $and: [
```
Comment (Nainterceptor) [inline routes/pois.mjs]: anti-pattern right here

File: routes/pois.mjs [hunk 3]
```diff
@@ -30,8 +30,33 @@ const router = generateCRUD(Poi, {
...
+        if (!enabled || enabled === 'false') {
+          ctx.filters.enabled = { $ne: false };
+        } else {
+          delete ctx.filters.enabled;
+        }
+        await next();
+      },
+      async (ctx, next) => {
+        const filters = Object.keys(ctx.filters).map((key) => ({
+          [key]: ctx.filters[key],
```
Comment (Nainterceptor) [inline routes/pois.mjs]: ? Read it again, it's a funny code.
You're iterating over an object to copy every key and values.
It's look like a simple 
```
const filters = ctx.filters
```

Comment (Nainterceptor) [inline routes/pois.mjs]: ? Read it again, it's a funny code.
You're iterating over an object to copy every key and values.
It's look like a simple 
```
const filters = ctx.filters
```

File: routes/pois.mjs [hunk 4]
```diff
@@ -30,8 +30,33 @@ const router = generateCRUD(Poi, {
     right: [CAN_LIST_POI, CAN_LIST_POI_LOCAL],
     filters: {
       campus: 'campus._id',
+      withDisabled: 'enabled',
     },
     middlewares: [
+      async (ctx, next) => {
+        const { enabled } = ctx.filters;
+        if (!enabled || enabled === 'false') {
+          ctx.filters.enabled = { $ne: false };
```
Comment (Nainterceptor) [inline routes/pois.mjs]: anti-pattern here

File: models/poi.mjs [hunk 5]
```diff
@@ -31,6 +35,14 @@ PoiSchema.virtual('campus.id')
     this.campus._id = id;
   });
 
+PoiSchema.statics.processDocumentsToAddEnable = async function () {
+  const docs = await this.find();
+  await Promise.all(docs.map(async (doc) => {
+    if (doc.enabled) {
+      await this.findByIdAndUpdate(doc._id, { enabled: true });
+    }
+  }));
```
Comment (Nainterceptor) [inline models/poi.mjs]: Ok, we don't want this feature, it's better to query with $or to query documents with enabled: true or not $exists.
BUT, for your information, you can run directly and update over the whole collection to change every document in a single query.

File: routes/pois.mjs (hunk 4 above)
Comment (Scttpr) [reply routes/pois.mjs]: Need more insights about anti-pattern here, I have a fix but it's more a "move away" rather than a pattern change :/

**Pull request merged**